ingredient/ingredient-postgresql/src/plugin.ts

```diff
@@ -1,28 +1,145 @@
+            params.virtualNetworkDeploymentName = {value: `virtualNetwork_${timestamp}`};
+            params.virtualNetworkLinkDeploymentName = {value: `virtualNetworkLink_${timestamp}`};
+            params.privateDnsZoneDeploymentName = {value: `privateDnsZone_${timestamp}`};
+            
+            // Hard coding this for security
+            params.publicNetworkAccess = {value: 'Disabled'};
+        }
+
+        if (!params.firewallRules)
+        {
+            params.firewallRules = {value: {rules: [] }};
+        }
 
-            await this._helper.DeployTemplate(this._name, ARMTemplate, params, await util.resource_group())
+        this.trimParametersForARM(params);
+
+        try {
+            let util = IngredientManager.getIngredientFunction("coreutils", this._ctx);
+            this._logger.log('PostgreSQL Plugin Logging: ' + this._ingredient.properties.parameters)
+
+            await this._helper.DeployTemplate(this._name, this._armTemplate, params, await util.resource_group())
 
         } catch(error){
-            this._logger.error('deployment failed: ' + error)
+            this._logger.error('Deployment failed: ' + error)
             throw error
         }
     }
-}
\ No newline at end of file
+
+    private async getVnetData(params: any): Promise<VnetData> {
+        //var data: any = {}};
+        let util = IngredientManager.getIngredientFunction("coreutils", this._ctx);
+
+        let vNet: VirtualNetwork = await this._functions.get_vnet(params.virtualNetworkResourceGroup.value, params.virtualNetworkName.value)
+        let subnetPropertiesGet: Subnet = await this._functions.get_subnet(params.virtualNetworkResourceGroup.value, params.virtualNetworkName.value, params.subnetName.value)
+        let privateDnsZoneName = this._functions.create_resource_uri(this._access); // todo investigate public equivalent.
+        let dnsZone = await this._functions.get_private_dns_zone(params.virtualNetworkResourceGroup.value, privateDnsZoneName)
+        let dnsZoneIsNew: boolean = false;
+
+        // if dnsZone doesn't exist, generate its id
+        if (dnsZone === undefined)
+        {
+            dnsZoneIsNew = true;
+            dnsZone = { id: `/subscriptions/${this._ctx.Environment.authentication.subscriptionId}/resourceGroups/${params.virtualNetworkResourceGroup.value}` +
+                `/providers/Microsoft.Network/privateDnsZones/${privateDnsZoneName}`};
+        }
+        
+        let vnetData: VnetData = {
+            value: {
+                virtualNetworkName: params.virtualNetworkName.value,
+                virtualNetworkId: vNet.id!,
+                subnetName: params.subnetName.value,
+                virtualNetworkAddressPrefix: subnetPropertiesGet.addressPrefix!,
+                virtualNetworkResourceGroupName: params.virtualNetworkResourceGroup.value,
+                location: await util.current_region(), // maybe primary_region()?
```

Comment:
Maybe! I don't know. Why don't you tell me? :)

Reply:
Actually there is a third and better option. `vNet.location!`. Not sure why I didn't use that in the first place...